File: python/7 kyu/231.py
# https://www.codewars.com/kata/56e56756404bb1c950000992/train/python

# Passed

from math import gcd

def sum_differences_between_products_and_LCMs(pairs):
    result = 0
    for pair in pairs:
        num1, num2 = pair
        gcd_value = gcd(num1, num2)
        lcm = (num1 / gcd_value) * (num2 / gcd_value) * gcd_value if gcd_value != 0 else 0
        result += (num1 * num2 - lcm)
        
    return result

output = sum_differences_between_products_and_LCMs([[15,18], [4,5], [12,60]])
print(output)

File: nodejs/7 kyu/231.js
// https://www.codewars.com/kata/57212c55b6fa235edc0002a2/train/javascript

// Passed

function commonGround(s1, s2) {
  let s2Words = s2.split(" ");
  let commonWords = [];
  for (const s2Word of s2Words) {
    if (s1.includes(s2Word)) {
      commonWords.push(s2Word);
    }
  }

  let result = commonWords.length ? commonWords.join(" ") : "death";
  return result;
}

const output = commonGround("eat a burger and drink a coke", "drink a coke");
console.log(output);